module-4/backend/week_two.md

```diff
@@ -0,0 +1,45 @@
+## Week Two - Module 4 Recap
+
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+
+Note: When you're done, submit a PR. 
+
+1. What is Webpack and why is it useful?
+- Wepback is a package for node projects that is similar to the asset pipeline in Rails. It looks at your asset files in your project (JS, CSS etc) and packages them up into one file per asset (so one js, one css) for production. There is a difference between what we as developers want/use to make our lives easier - multiple files, white space etc. and what the broswer wants - one file, no white space and webpack takes what we have written and makes it easier for the browser to read it. 
+
+2. When do you want to use event delegation?
+- Event delegation is good in cases where you might have memory leak - you tied an event listener to something that is no longer in the DOM and so the program is "listening" for something that will never happen and wasting memory use on it. Event delegation ties the listener to the parent object that then analyzes the event to see if it matches exactly what you are looking for. 
+
+3. What's one difference between ES5 and ES6?
+- arrow notation vs. function notation; var vs. let and const; string interpretation 
+
+4. How are you using the MVC design pattern in your Quantified Self project?
+- we have broken out models for foods and meals and are planning to break out controllers to help organize our code better instead of one big file. 
+
+5. How do you execute raw SQL in node?
+- knex package lets you use knex.raw and then pass in a raw SQL query
+
+6. What's a callback function and what are some reasons when we use/need callback functions?
+- a callback function is a function passed to another function and then executed inside that function. They help with flow control - aschronicity of JS
+
+7. What is CORS?
+- it is a library/package? (not sure what to call it) that allows sites(or different domains) to communicate with eachother 
+
+8. What are some steps to avoid CORS?
+- unsure - had to google this - looks like you could use document.domain or window.postMessage? 
```

Comment:
there are a few different work arounds in order to get around CORS. you can use a Express CORS middleware or a Rack CORS gem depending on where you're experiencing the error. check back in the lesson plan Friday AM 👍